### PR TITLE
Implement v2 styles endpoint and update layer inheritance

### PR DESCRIPTION
**Pull request by gowin20**
Implements #150 #174 #170

### #150 - Change VectorBasemapLayer to inherit from VectorTileLayer
- VectorBasemapLayer now inherits `onAdd` and `_createMaplibreLayer` from VectorTileLayer, and performs a `super` call to the VectorTileLayer constructor.
- Some functions remain unmerged, such as `onRemove` and `_asyncAdd`, due to the dynamic data attribution in VectorBasemapLayer

### #174 - add support for the v2 basemap styles service
(see #175)
* Added new `getBasemapStyleV2Url` function that returns a formatted reference to the new v2 endpoint
* New `version` parameter in `VectorBasemapLayer` that specifies the endpoint to access (1 or 2) If not provided, the version will attempt to set itself based on the provided style enumeration
* New `language` parameter that works with v2 of the service, accepts valid [language parameters](https://developers.arcgis.com/rest/basemap-styles/arcgis-outdoor-style-get/#language)

### #170 - fix default pane for VectorBasemapLayer
(see #181)
* Fixed a bug where VectorBasemapLayer was not being added to the `tilePane`
* Added additional code to create a new `esri-details` pane for styles of type `:Details`/`/details`
* Changed default z-index of `esri-labels` pane to display below the `overlayPane`
* Added unit tests

### CODE REVIEW

File: spec/VectorBasemapLayerSpec.js [hunk 1]
```diff
@@ -43,13 +46,144 @@ describe('VectorBasemapLayer', function () {
...
+  it('should error if a language is provided when loading a custom basemap style', function () {
+    expect(function () {
+      L.esri.Vector.vectorBasemapLayer(customBasemap, {
+        apikey,apikey,
+        version:2,
+        language:language
+      })
+    }).to.throw('The \'language\' parameter is not supported for custom basemap styles')
+  })
+
```
Comment (gavinr) [inline spec/VectorBasemapLayerSpec.js:186]: I don't see a unit test for the new functionality "it should load a v2 basemap if using a v2 basemap key enum even if not passing the `version: 2` property" - can you add a unit test for that case?

File: src/VectorTileLayer.js [hunk 2]
```diff
@@ -28,10 +24,12 @@ export var VectorTileLayer = Layer.extend({
       this.options.apikey = this.options.apiKey;
     }
 
-    // if apiKey is passed in, use it as a token
-    // (opposite from VectorBasemapLayer.js)
+    // if apiKey is passed in, propogate to token
+    // if token is passed in, propogate to apikey
```
Comment (gavinr) [inline src/VectorTileLayer.js:28]:
```suggestion
    // if apiKey is passed in, propagate to token
    // if token is passed in, propagate to apiKey
```

**Review by gavinr — APPROVED** [review]
Overall seems to work for me. A few small things:

File: src/Util.js [hunk 3]
```diff
@@ -5,17 +5,47 @@ import { request, Support, Util } from 'esri-leaflet';
...
   return url;
 }
 
+export function getBasemapStyleV2Url (style, apikey, language) {
+  if (style.includes(':')) {
+    throw new Error(style + ' is a v1 style enumeration. Set version:1 to request this style');
+  }
+
+  let url = 'https://basemapstyles-api.arcgis.com/arcgis/rest/services/styles/v2/styles/';
+  if (!(style.indexOf('osm/') === 0 || style.indexOf('arcgis/') === 0) && style.length === 32) {
```
Comment (patrickarlt) [inline src/Util.js:29]:
```suggestion
  if (!(style.startsWith('osm/') || style.startsWith('arcgis/')) && style.length === 32) {
```

**Review by patrickarlt — APPROVED** [review]
LGTM! Minor code suggestion but please add the test that @gavinr has indicated.

**Pull request merged**